metatron/src/test/java/uk/gov/ida/eidas/metatron/apprule/rules/MetatronTests.java

```diff
@@ -0,0 +1,263 @@
+package uk.gov.ida.eidas.metatron.apprule.rules;
+
+import io.dropwizard.client.JerseyClientBuilder;
+import org.glassfish.jersey.client.ClientProperties;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+import org.opensaml.saml.saml2.metadata.EntityDescriptor;
+import org.opensaml.saml.saml2.metadata.KeyDescriptor;
+import org.opensaml.saml.saml2.metadata.SPSSODescriptor;
+import org.opensaml.xmlsec.signature.Signature;
+import uk.gov.ida.eidas.metatron.resources.MetatronResource;
+import uk.gov.ida.saml.core.test.OpenSAMLRunner;
+import uk.gov.ida.saml.core.test.TestCredentialFactory;
+import uk.gov.ida.saml.core.test.builders.SignatureBuilder;
+import uk.gov.ida.saml.core.test.builders.metadata.EntityDescriptorBuilder;
+import uk.gov.ida.saml.core.test.builders.metadata.KeyDescriptorBuilder;
+import uk.gov.ida.saml.core.test.builders.metadata.SPSSODescriptorBuilder;
+import uk.gov.ida.saml.metadata.test.factories.metadata.MetadataFactory;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Supplier;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.METADATA_SIGNING_A_PRIVATE_KEY;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.METADATA_SIGNING_A_PUBLIC_CERT;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PUBLIC_ENCRYPTION_CERT;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PUBLIC_SIGNING_CERT;
+
+@RunWith(OpenSAMLRunner.class)
+public class MetatronTests {
+    public static TestMetadataResource testMetadataResource;
+    public static CreatesConfigDropwizardClientRule testMetadataServer;
+    public static MetatronAppRule metatronAppRule;
+    private static Client client;
+
+    @ClassRule
+    public static RuleChain ruleChain;
+
+    static {
+        try {
+            ruleChain = init();
+        } catch (Exception e) {
+            Assert.fail(e.getMessage());
+        }
+    }
+
+    public static RuleChain init() throws Exception {
+        testMetadataResource = new TestMetadataResource();
+        testMetadataServer =  new CreatesConfigDropwizardClientRule(testMetadataResource);
+        waitWhile(() -> testMetadataServer.getPort() == 0, "Waiting for client rule port ");
+
+        testMetadataResource.init(testMetadataServer.getPort());
+        metatronAppRule = new MetatronAppRule();
+
+        return RuleChain.outerRule(testMetadataServer).around(metatronAppRule);
+    }
+
+    private static void waitWhile(Supplier<Boolean> condition, String message) {
+        // This needs a time-out mechanism
```

Comment:
Could create a datetime, and then compare the current datetime to it in the loop. If it's more than 1 second or two, bail out?

Reply:
Also, system out printing doesn't appear anywhere that I can see, could drop it?